lib/matplotlib/backends/backend_pdf.py

```diff
@@ -2523,18 +2523,21 @@ def savefig(self, figure=None, **kwargs):
             instance is provided, this figure is saved. If an int is specified,
             the figure instance to save is looked up by number.
         """
-        if isinstance(figure, Figure):
-            figure.savefig(self, format='pdf', **kwargs)
-        else:
+        if not isinstance(figure, Figure):
             if figure is None:
-                figureManager = Gcf.get_active()
-            else:
-                figureManager = Gcf.get_fig_manager(figure)
-            if figureManager is None:
-                raise ValueError("No such figure: " + repr(figure))
+                manager = Gcf.get_active()
             else:
-                figureManager.canvas.figure.savefig(self, format='pdf',
-                                                    **kwargs)
+                manager = Gcf.get_fig_manager(figure)
+            if manager is None:
+                raise ValueError("No figure {}".format(figure))
+            figure = manager.canvas.figure
+        # Force use of pdf backend, as PdfPages is tightly coupled with it.
+        try:
+            orig_canvas = figure.canvas
+            figure.canvas = FigureCanvasPdf(figure)
+            figure.savefig(self, format="pdf", **kwargs)
```

Comment:
What if we were already using the pdf backend?  How much extra work would this create?  Does the new canvas need to get explicitly destroyed to free memory?

Reply:
Creating the FigureCanvasPdf is really cheap (it is defined just below -- its init is directly inherited from FigureCanvasBase, which doesn't do anything fancy either).
The temporary canvas will get GC'd per normal python semantics.